manifest/schema2/manifest.go

```diff
@@ -48,6 +52,14 @@ func init() {
 	}
 }
 
+// LayerDescriptor is an extension to distribution.Descriptor that
+// has additional fields specific to layers.
+type LayerDescriptor struct {
```

Comment:
I'm not sure about the value of defining `LayerDescriptor` separately. `URLs` would be just as valid for any resource. Let's just add this to `Descriptor`.